src/main/java/org/threadly/litesockets/utils/TransactionalByteBuffers.java

```diff
@@ -0,0 +1,208 @@
+package org.threadly.litesockets.utils;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayDeque;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class TransactionalByteBuffers extends MergedByteBuffers {
+  private static final String ACCESS_ERROR = "Can not call method from different thread then the transaction begain with";
+  private final ReentrantLock lock = new ReentrantLock();
+  private ArrayDeque<ByteBuffer> consumedBuffers = new ArrayDeque<ByteBuffer>(8); 
+  private int consumedSinceBegin = 0;
+  
+  /**
+   * This mark the beginning of a new transaction.  anything done from this point can either 
+   * be committed or rolled back and forgotten.
+   * 
+   * If an operation is done with out using begin it is committed immediately on completion.
+   * 
+   */
+  public void begin() {
+    lock.lock();
+    consumedSinceBegin = 0;
+    consumedBuffers.clear();
+  }
+  
+  /**
+   * Commit a already started transaction.  Until this is done anything pulled out of the consolidator can be reverted.
+   * once this happens all changes are permanent and forever.
+   * 
+   * 
+   */
+  public void commit() {
+    if(!lock.isLocked()) {
+      return;
+    }
+    if (! lock.isHeldByCurrentThread()) {
+      throw new IllegalStateException("Must call by same Thread as begin!");
+    }
+    consumedSinceBegin = 0;
+    consumedBuffers.clear();
+    lock.unlock();
+  }
+  
+  /**
+   * This allows you to roll back all operations done on the consolidator since the begin was called.
+   * 
+   *  NOTE:  If you are modifying the data in the underlying byteArrays that will not be reverted
```

Comment:
I would also include this note in a description about the class (which is currently missing)...it's pretty important